### ISSUE TITLE
[REFACTOR] nn.Module을 상속하는 Loss Class를 반환하도록 수정하기

### ISSUE THREAD

**Issue by rudeuns** [boostcampaitech6/level1-imageclassification-cv-14#42]
## Background
- nn.Module을 상속하지 않는 손실함수는 requires_grad=True로 만들도록 구현이 어려움
- nn.Module을 상속하지 않으면 loss.backward가 진행되지 않음
- 클래스로 구현을 한 뒤 함수에서 클래스를 반환하도록 코드 수정이 필요

## To do
- [x] nn.Module을 상속하도록 Loss Class 구현
- [x] Loss Class를 반환하는 loss func으로 수정
